typings/config.d.ts

```diff
@@ -52,6 +64,36 @@ declare global {
         instance: InstanceType<typeof Gatherer>;
         options?: {};
       } | string;
+
+      export interface CategoryJson {
+        name: string;
+        description: string;
+        audits: CategoryMemberJson[];
+      }
+
+      export interface GroupJson {
+        title: string;
+        description: string;
+      }
+
+      /**
+       * A LH Artifacts object, but the traces and devtoolsLogs are replaced
+       * with file paths of json files to import as those artifacts.
+       */
+      export interface ArtifactsJson extends Omit<Artifacts, 'traces'|'devtoolsLogs'> {
+        traces: Record<string, string>;
+        devtoolsLogs: Record<string, string>;
+      }
+
+      /**
+       * Reference to an audit member of a category and how its score should be
+       * weighted and how its results grouped with other members.
+       */
+      export interface CategoryMemberJson {
```

Comment:
same as `LH.Result.Category.CategoryMember`

can we avoid defining it twice?

Reply:
> can we avoid defining it twice?

this is mostly mental organization that can be cleaned up when we add types to `config.js`. I'm not sure what we'll need by the end of things, so it was easier to keep separate for now.

Even if they end up exactly the same, we may still want two separate type names, one for acceptable input to `config.js` and one for what comes out. @patrickhulce mentions he uses something like `typename` and `typenameUnsafe`, and something like that might make it more obvious what the difference is. Though hopefully in our case you'll never see a `*Json` type inside of anything called by runner

Reply:
okay gotcha. can we do a comment like

> Object shape currently matches LH.Result.Category.CategoryMember

Reply:
added a comment and TODO

Reply:
okay. I think the resolution here is that we ideally split Config and ConfigJSON into two `.d.ts` files. (and I'd favor a clearer module split for all the JSON buddies to keep them separate from what `new Config()` returns. \o/